src/Dialect/ONNX/ElementsAttrBuilder.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===----------------------- ElementsAttrBuilder.hpp ----------------------===//
+//
+// ElementsAttrBuilder builds DisposableElementsAttr instances.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "src/Dialect/ONNX/DisposableElementsAttr.hpp"
+#include "src/Dialect/ONNX/DisposablePool.hpp"
+#include "src/Support/Strides.hpp"
+#include "src/Support/WideNum.hpp"
+
+#include <atomic>
+
+namespace onnx_mlir {
+
+// Builds DisposableElementsAttr instances.
+// Every instance is inserted into DisposablePool which garbage collects
+// unreachable instances between compiler passes.
+class ElementsAttrBuilder {
+public:
+  ElementsAttrBuilder(DisposablePool &disposablePool);
```

Comment:
are the two constructor identical? just different objects, or there is a difference between the two. I would specify either way in a comment

Reply:
I changed it to have only one constructor